src/Castle.Windsor.Extensions.DependencyInjection/WindsorServiceProviderFactoryBase.cs

```diff
@@ -56,6 +56,12 @@ protected virtual void SetRootContainer(IWindsorContainer container)
 		{
 			rootContainer = container;
 			AddSubSystemToContainer(rootContainer);
+			var rootScope = CreateRootScope();
+			
+			container.Register(Component
```

Comment:
Registering an `ILifetimeScope` in the container is likely going to cause all sorts of pain.